### DISCUSSION TITLE
Threshold

### DISCUSSION THREAD

**Discussion by Asd-g** [HomeOfAviSynthPlusEvolution/neo_f3kdb#16] [General]
@msg7086, what do you think to change how the threshold is scaled? [Left shift by 7](https://github.com/HomeOfAviSynthPlusEvolution/neo_f3kdb/blob/master/src/f3kdb.hpp#L171-L173) gives almost whole 16-bit range and keeps the current threshold limit [0,511].

**Comment by msg7086** [comment]
I don't quite get the context. What's the intention?

**Comment by Asd-g** [comment]
The intention is to expand the threshold effect. Currently there are 31 ranges and they can be 255.

**Reply by msg7086** [reply]
Make sense, sound good to me.

**Comment by Asd-g** [comment]
> Currently there are 31 ranges and they can be 255.

I meant threshold ranges not the parameter `ranges`.

Take `sample_mode=2, blur_first=true`. Now threshold values between 1 and 16 return same output, threshold values between 17 and 32 return same output and so on... If the threshold is left shift by 5 instead of 2 - threshold values 1 and 2 return same output, threshold values 3 and 4 return same output and so on.

 Take `blur_first=false`. Now threshold values between 1 and 64 return same output, threshold values between 65 and 128 return same output and so on. If the threshold is left shift by 5 instead of 2 - threshold values between 1 and 8 return same output, threshold between 9 and 16 return same output and so on.

The down side would be that the default threshold values have to be changed. Also the existing scripts using the filter have to be changed for the new thresholds.

**Reply by msg7086** [reply]
> Also the existing scripts using the filter have to be changed for the new thresholds.

Better introducing new parameters as we don't want to break for everyone.

Try something like y2 cb2 cr2, default to -1, and do something like y2>=0?y2<<shift2:y<<shift.
